src/Dependencies/Collections/SegmentedArray`1+PrivateMarshal.cs

```diff
@@ -14,5 +16,17 @@ internal static class PrivateMarshal
         /// <inheritdoc cref="SegmentedCollectionsMarshal.AsSegments{T}(SegmentedArray{T})"/>
         public static T[][] AsSegments(SegmentedArray<T> array)
             => array._items;
+
+        public static SegmentedArray<T> AsSegmentedArray(T[][] segments)
+        {
+            if (segments is null)
+                throw new ArgumentNullException(nameof(segments));
+
+            var length = 0;
+            foreach (var segment in segments)
+                length += segment.Length;
```

Comment:
All segments except for the last segment have a known size. The length of the array should be calculated by adding the final segment length to the a multiple of the known segment size.

I'm undecided on whether argument validation should go here or in the caller. I lean towards placing it exclusively in the caller, and potentially omitting it (or making it debug assertions) since this is an unsafe class.

Reply:
Missed this one earlier. Removed the arg check, as you mentioned this is unsafe code, you should know what you're doing if you use this.

Reply:
➡️ Now resolved